R/data.table.R

```diff
@@ -1849,7 +1849,7 @@ replace_dot_alias = function(e) {
     if (use.I) assign(".I", seq_len(nrow(x)), thisEnv)
     ans = gforce(thisEnv, jsub, o__, f__, len__, irows) # irows needed for #971.
     gi = if (length(o__)) o__[f__] else f__
-    g = lapply(grpcols, function(i) groups[[i]][gi])
+    g = lapply(grpcols, function(i) .Call(CsubsetVector, groups[[i]], gi)) # use CsubsetVector instead of [ since this preserves attributes #5567
```

Comment:
Would be nice to measure impact on small and big inputs here.

Reply:
I thought the same. We could ofc copy attributes if present on `R` level but that might get slow for many groups?